bot/exts/fun/madlibs.py

```diff
@@ -0,0 +1,116 @@
+import json
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+from typing import TypedDict
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+TIMEOUT = 60.0
+
+
+class MadlibsTemplate(TypedDict):
+    """Structure of a template in the madlibs JSON file."""
+
+    title: str
+    blanks: list[str]
+    value: list[str]
+
+
+class Madlibs(commands.Cog):
+    """Cog for the Madlibs game."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.templates = self._load_templates()
+
+    @staticmethod
+    def _load_templates() -> list[MadlibsTemplate]:
+        madlibs_stories = Path("bot/resources/fun/madlibs_templates.json")
+
+        with open(madlibs_stories) as file:
+            return json.load(file)["templates"]
+
+    @staticmethod
+    def madlibs_embed(part_of_speech: str, number_of_inputs: int) -> discord.Embed:
+        """Method to have the bot send an embed with the game information."""
+        madlibs_embed = discord.Embed(
+            title="Madlibs",
+            color=Colours.python_blue,
+        )
+
+        madlibs_embed.add_field(
+            name="Enter a word that fits the given part of speech!",
+            value=f"Part of speech: {part_of_speech}"
+        )
+
+        madlibs_embed.set_footer(text=f"Inputs remaining: {number_of_inputs}")
+
+        return madlibs_embed
+
+    @commands.command()
+    async def madlibs(
+        self,
+        ctx: commands.Context,
+    ) -> None:
```

Comment:
This should also fit in one line, so the separating isn't necessary.
```suggestion
    async def madlibs(self, ctx: commands.Context) -> None:
```